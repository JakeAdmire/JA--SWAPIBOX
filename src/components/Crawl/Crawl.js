import React from 'react';

export function Crawl({crawl, title, date}) {
  return (
    <div>
      <textarea readOnly value={crawl}></textarea>
      <p>{title}</p>
      <p>{date}</p>
    </div>
  )
}