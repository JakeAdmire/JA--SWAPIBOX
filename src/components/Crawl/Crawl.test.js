import { enzyme, shallow } from 'enzyme';
import React from 'react'
import { Crawl } from './Crawl';

describe('Crawl', () => {

  let wrapper;

  let crawlContent = {
    opening_crawl: 'Once upon a time', 
    title: 'Movie1', 
    release_date: '3-28-2019'
  }

  wrapper = shallow(
    <Crawl {...crawlContent} />
  )

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }) 

})