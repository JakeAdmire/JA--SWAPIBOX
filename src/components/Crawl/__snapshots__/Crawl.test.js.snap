// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Crawl
    opening_crawl="Once upon a time"
    release_date="3-28-2019"
    title="Movie1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          readOnly={true}
          value="Once upon a time"
        />,
        <p>
          Movie1
        </p>,
        <p>
          3-28-2019
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "readOnly": true,
          "value": "Once upon a time",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Movie1",
        },
        "ref": null,
        "rendered": "Movie1",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3-28-2019",
        },
        "ref": null,
        "rendered": "3-28-2019",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            readOnly={true}
            value="Once upon a time"
          />,
          <p>
            Movie1
          </p>,
          <p>
            3-28-2019
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "readOnly": true,
            "value": "Once upon a time",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie1",
          },
          "ref": null,
          "rendered": "Movie1",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3-28-2019",
          },
          "ref": null,
          "rendered": "3-28-2019",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
